packages/sales-api-service/src/services/recurring-payments.service.js

```diff
@@ -0,0 +1,54 @@
+import { findByExample, findById, Permission, RecurringPayment } from '@defra-fish/dynamics-lib'
+
+export const getRecurringPayments = async () => {
+  const currentDate = new Date()
+  const dueDates = []
+  for (let i = 0; i <= 10; i += 2) {
+    const date = new Date(currentDate)
+    date.setDate(currentDate.getDate() - i)
+    dueDates.push(date)
+  }
+
+  const recurringPayments = []
+  for (const dueDate of dueDates) {
+    const activeRecurringPayments = await findByExample(new RecurringPayment())
+    const dueRecurringPayments = await findRecurringPaymentsInDateRange(activeRecurringPayments, dueDate)
+
+    if (dueRecurringPayments.length !== 0) {
+      recurringPayments.push(dueRecurringPayments)
```

Comment:
This is not the way you should concatenate arrays - push is for pushing a single item onto an array. Have a look at this codepen: https://codepen.io/jaucourt/pen/YzJmQWV. You'll see that when you push one array onto another, it's added as an array within an array.

One way to concatenate arrays is to use the [concat function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat). This probably isn't the most performant, though, so you need to explore how best to do it.